internal/layerstate/states.go

```diff
@@ -5,12 +5,20 @@ import (
 	"errors"
 )
 
+type InstanceStatus string
+
+const (
+	InstanceStatusReady InstanceStatus = InstanceStatus("ready")
```

Comment:
Do you prefer `ready` or `success`? I feel like `success` is a better counterpart to `fail`.

Nbd though

Reply:
I don't see this `ready` status as a counterpart to `fail`. I think the "counterpart" of `ready` is like `killed` or `dead`, maybe we could go with `alive`?
The `fail` is a state in the middle, perhaps `fail` is also bad.

I tried to draw a graph and ended up with this, I'll use new names for all status:

```mermaid
  graph TD;
      spawning-->alive;
      spawning-->broken;
      alive-->refreshing;
      alive-->killing;
      refreshing-->alive;
      refreshing-->broken;
      broken-->refreshing;
      broken-->killing;
      killing-->broken;
      killing-->dead;
```

Btw, I think I like the new names 🤔 

Also I think we should eventually implement tracking of all these status.

Reply:
Lgtm, but let's go with "faulty" instead of broken.